MfciPkg/MfciDxe/MfciDxe.c

```diff
@@ -31,6 +31,27 @@
 
 #include "MfciDxe.h"
 
+/**
+ * the following helps iterate over the functions and set the corresponding target variable names
+ */
+
+// define a structure that pairs up the function pointer with the UEFI variable name
+typedef struct {
+  MFCI_DEVICE_ID_FN    DeviceIdFn;
+  CHAR16               *DeviceIdVarName;
+} MFCI_DEVICE_ID_FN_TO_VAR_NAME_MAP;
+
+// populate the array of structures that pair up the functions with variable names
+#define MFCI_TARGET_VAR_COUNT  5

```

Comment:
Unless `MFCI_TARGET_VAR_COUNT` is used elsewhere, I think it it'd be easier to maintain the code without it since the array size can be deduced from the initializers here.

Reply:
Yeah, hold on this PR. There seems to be some complication with other platforms...

Reply:
The count is in place so as ARRAY_SIZE will not throw an exception on sizeof returning 0.

Reply:
I see it became an extern after so that makes sense in that case.